### PR TITLE
TDL-17879 added authentication before discovery mode and unittests

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
- Added credentials in discovery mode.
- Throws an error if the creds are not correct.
- Added unittests for the same

# Manual QA steps
 - Check that with wrong credentials the discovery mode does not create a catalog file.
 - Check that with correct credentials the discovery mode creates correct catalog file.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_auth_in_discovery.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    return response
+
+@mock.patch('tap_stripe.utils.parse_args')
+class TestBasicAuthInDiscoverMode(unittest.TestCase):
+    @mock.patch('tap_stripe.discover')
+    @mock.patch('stripe.http_client.requests.Session.request')
+    def test_basic_auth_no_access_401(self, mock_request, mocked_discover, mocked_args):
+        '''
+            Verify exception is raised for no access(401) error code for basic auth
+            and discover is called once for setup Context.
```
Comment (savan-chovatiya) [inline tests/unittests/test_auth_in_discovery.py:32]: Update comment here as discover function is not called for this test.

File: tests/unittests/test_auth_in_discovery.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            self.assertEquals(str(e), expected_error_message)
+        # Verify that the discover is not called  when incorrect credentials are passed
+        self.assertEqual(mocked_discover.call_count, 0)
+
+    @mock.patch('tap_stripe.discover', return_value = {})
+    @mock.patch('stripe.http_client.requests.Session.request')
+    def test_basic_auth_access_200(self, mock_retrieve, mocked_discover, mocked_args):
+        '''
+            Verify discover mode is called if basic auth credentials are valid
+            and discover function is called twice for setup Context and discover mode.
```
Comment (savan-chovatiya) [inline tests/unittests/test_auth_in_discovery.py:50]: Update comment here as discover function is called only once for this test.

Comment (namrata270998) [reply tests/unittests/test_auth_in_discovery.py]: Updated

File: tests/unittests/test_auth_in_discovery.py (hunk 1 above)
Comment (namrata270998) [reply tests/unittests/test_auth_in_discovery.py]: Updated

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

File: tap_stripe/__init__.py [hunk 3]
```diff
@@ -894,6 +894,10 @@ def sync():
 def main():
     # Parse command line arguments
     args = utils.parse_args(REQUIRED_CONFIG_KEYS)
+    # set the config and state in prior to check te authentication in the discovery mode itself.
```
Comment (KrisPersonal) [inline tap_stripe/__init__.py:897]: Spelling error in Comment. Please correct it

**Review by kspeer825 — APPROVED** [review]

**Review by RushT007 — APPROVED** [review]
Looks good to me!

File: tap_stripe/__init__.py (hunk 3 above)
Comment (namrata270998) [reply tap_stripe/__init__.py]: Corrected the mistake

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**